amundsen_application/static/js/ducks/search/reducer.ts

```diff
@@ -104,6 +104,7 @@ export default function reducer(state: SearchReducerState = initialState, action
     case SearchResource.FAILURE:
       return {
         ...initialState,
+        search_term: (<SearchAllRequest>action).payload.term,
```

Comment:
This doesn't look quite right. Two different cases are now executing this logic, yet they don't have the same types. The action types are `SearchAllResponse` and `SearchResourceResponse`.  Both of these have a payload that is will have to contains all of these terms as well:
```
payload: {
    search_term: string;
    isLoading: boolean;
    dashboards: DashboardSearchResults;
    tables: TableSearchResults;
    users: UserSearchResults;
  };
```

I think your changes in the `SearchBar` look good, however now that we've discovered what the necessary changes are for the `category:fake` case I think it makes more sense to add it in as part of #201. You're free to invest the effort, but because I will still have to tweak it anyway it could save you some time.

Reply:
I've changed the call here, lmk if that looks correct - I feel like it makes sense to grab the search_term from SearchAllResponse's payload